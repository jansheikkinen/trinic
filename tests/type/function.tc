// Function

function main(void) -> void do
  let a: (void) -> void, b: (int8, int8) -> void, c: (void) -> int8,
      d: (T, V) -> for T + V: trait Add + trait Display = undefined;
end
