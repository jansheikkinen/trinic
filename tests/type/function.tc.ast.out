
### AST GENERATION ###
Declaration FUNCTION  main
  Stmt VARDECL
    Arg IDENTIFIER
      Identifier_Arg a
        Type FUNCTION
      Identifier_Arg b
        Type FUNCTION
          Arg SUMARGTYPE
            Sum_Arg_Type 
              Type TYPE INT8
            Sum_Arg_Type 
              Type TYPE INT8
      Identifier_Arg c
        Type FUNCTION
          Type TYPE INT8
      Identifier_Arg d
        Type FUNCTION
          Arg SUMARGTYPE
            Sum_Arg_Type T
            Sum_Arg_Type V
          Type IDENTIFIER T
          Arg GENERIC_DEFS
            Generic_Def
              Arg GENERIC_DEFL
                IDENTIFIER T
                IDENTIFIER V
              Arg GENERIC_DEFR
                Type TRAIT Add
                Type TRAIT Display
    Arg EXPR
      Expr UNDEFINED
